x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtDateClaimValidatorTests.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import org.elasticsearch.ElasticsearchSecurityException;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
+
+import java.text.ParseException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class JwtDateClaimValidatorTests extends ESTestCase {
+    private Clock clock;
+
+    @Before
+    public void init() {
+        clock = mock(Clock.class);
+    }
+
+    public void testClaimIsNotDate() throws ParseException {
+        final String claimName = randomAlphaOfLengthBetween(10, 18);
+        final JwtDateClaimValidator validator = new JwtDateClaimValidator(
+            clock,
+            claimName,
+            TimeValue.ZERO,
+            randomFrom(JwtDateClaimValidator.Relationship.values()),
+            randomBoolean()
+        );
+
+        final SignedJWT jwt = prepareJwt(Map.of(claimName, randomAlphaOfLengthBetween(3, 8)));
+        final ElasticsearchSecurityException e = expectThrows(ElasticsearchSecurityException.class, () -> validator.validate(jwt));
+        assertThat(e.getMessage(), containsString("cannot parse date claim"));
+        assertThat(e.getCause(), instanceOf(ParseException.class));
+    }
+
+    public void testClaimDoesNotExist() throws ParseException {
+        final String claimName = randomFrom(randomAlphaOfLengthBetween(3, 8), "iat", "nbf", "auth_time");
+        final boolean allowNull = randomBoolean();
```

Comment:
I think this should be 2 tests, rather than a randomised test.

Reply:
Agreed. I splited it into 2.